test/src/test/java/org/jvnet/hudson/main/JenkinsRuleTimeoutTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2015 CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jvnet.hudson.main;
+
+import hudson.model.listeners.ItemListener;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.Rule;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.TestExtension;
+
+public class JenkinsRuleTimeoutTest {
+
+    @Rule public JenkinsRule r = new JenkinsRule();
+    {
+        r.timeout = 10; // let us not wait three minutes!
+    }
+
+    @Test public void hangInterruptiblyInTest() throws Exception {
+        try {
+            hangInterruptibly();
+        } catch (InterruptedException x) {
+            System.err.println("Interrupted, good.");
+        }
+    }
+
+    @Ignore("TODO ought to use the idiom from org.junit.internal.runners.statements.FailOnTimeout: run test in its own thread so we can fail even if it does not respond to interruptions")
```

Comment:
I completely agree that would be really useful and more manageable to have
this project out of Jenkins core project and move to its own repo and
artifact. Next week probably I have a meeting with  @kohsuke
<https://github.com/kohsuke> and we can talk about this too.

Alex

El dj., 20 ag. 2015 a les 20:41, Kanstantsin Shautsou (<
notifications@github.com>) va escriure:

> In test/src/test/java/org/jvnet/hudson/main/JenkinsRuleTimeoutTest.java
> <https://github.com/jenkinsci/jenkins/pull/1809#discussion_r37565786>:
>
> > +public class JenkinsRuleTimeoutTest {
> > +
> > +    @Rule public JenkinsRule r = new JenkinsRule();
> > +    {
> > +        r.timeout = 10; // let us not wait three minutes!
> > +    }
> > +
> > +    @Test public void hangInterruptiblyInTest() throws Exception {
> > +        try {
> > +            hangInterruptibly();
> > +        } catch (InterruptedException x) {
> > +            System.err.println("Interrupted, good.");
> > +        }
> > +    }
> > +
> > +    @Ignore("TODO ought to use the idiom from org.junit.internal.runners.statements.FailOnTimeout: run test in its own thread so we can fail even if it does not respond to interruptions")
>
> @jglick <https://github.com/jglick> those who are reading this PR on GH
> enforced to scroll. //cc @lanwen <https://github.com/lanwen>
> Should be not so difficult to fit in 120-130 lines
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/jenkins/pull/1809/files#r37565786>.
>